### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by m-roberts**
### API changes
* Assert OLED SPI bus is 0, 1
* Refactor buttons module so that Buttons class can be rendered nicely in the docs (no functional change)
* Keyboard module now serves a KeyboardButton, to better reflect its usage
* Added Miniscreen object, updated all references to make use of it

### Examples
* General improvements - renaming files/docs headers, adding more printing, etc.

### Docs
* API section is now more intuitively split into 4 categories -
    * "General pi-top functionality" --> "pi-top device"
    * "PMA" --> "PMA components" (anything that uses our ports specifically)
    * "Peripherals" --> "pi-top Peripheral Devices" (pulse, proto+)
    * "Helpers" --> "System Devices" (camera, mic, keyboard, mouse, etc.)
* Removed all table of contents from pages (except More)
* Corrected explanation of how to use the camera format
* Moved all references from `automodule` to `autoclass` (except pulse, which does not have a class to do this with)
* Removed `Indices and tables` from home page (moved index to more explanatory text at bottom of "More Information"
* Moved “Getting Started with PMA” to a PMA section under “Key Concepts”, to make way for more stuff we want to include, such as how the SDK handles images, etc.
* Added initial "advanced usage" section for Robotics (currently hidden)
* Add [paragraph-level markup](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#paragraph-level-markup) in docs to communicate method deprecation
* Moved pi-top 'components' inside Pitop to properties
* All references updated to "class" where possible, explaining that classes are explicit representations of either a device and component

### CODE REVIEW

File: docs/api_pitop_device.rst [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+Check out :ref:`Key Concepts: pi-top [4] Miniscreen<key_concepts:pi-top [4] Miniscreen>` for useful information about how this class works.
+
+Buttons
+-------
+
+.. image:: _static/miniscreen/pi-top_4_Front_BUTTONS.jpg
+
+You can write code to respond to these buttons being pressed and released.
+
+The buttons are easily accessed from the :class:`pitop.miniscreen.Miniscreen` object:
```
Comment (m-roberts) [inline docs/api_pitop_device.rst:54]: Make clearer

File: docs/api_pitop_device.rst [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+.. autoclass:: pitop.miniscreen.CancelButton
+    :inherited-members:
+
+
+OLED Display
+------------
+
+.. image:: _static/miniscreen/pi-top_4_Front_OLED.jpg
+
+You can write code to send pixel data to the miniscreen's OLED display.
```
Comment (m-roberts) [inline docs/api_pitop_device.rst:80]: C'mon man - more info

File: docs/api_pitop_device.rst [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+All of the OLED display functionality is available by using the :class:`pitop.miniscreen.Miniscreen` class directly.
```
Comment (m-roberts) [inline docs/api_pitop_device.rst:82]: Reword

File: docs/api_pitop_peripheral_devices.rst [hunk 4]
```diff
@@ -0,0 +1,103 @@
+===============================
+API - pi-top Peripheral Devices
+===============================
+
+pi-topPROTO+
+==================
+
+This module provides 2 classes - a simple way to use a pi-topPROTO+'s onboard ADC (analog-to-digital converter), and another use it as a distance sensor.
```
Comment (m-roberts) [inline docs/api_pitop_peripheral_devices.rst:8]:
```suggestion
This module provides 2 classes - a simple way to use a pi-topPROTO+'s onboard ADC (analog-to-digital converter), and another to use it as a distance sensor.
```

File: docs/api_pma_components.rst [hunk 5]
```diff
@@ -1,14 +1,32 @@
...
+================================================
+
+.. image:: _static/pma/foundation_kit/components_spread.jpg
+
+.. image:: _static/pma/robotics_kit/Alex.jpg
+
+
+The Foundation & Expansion Plates and all the parts included in the Foundation & Robotics Kit are known as the pi-top Maker Architecture (PMA).
+
+Each component has a Python class associated with it.
```
Comment (m-roberts) [inline docs/api_pma_components.rst:12]: Describe this better

**Review by jcapona — APPROVED** [review]

**Pull request merged**